base/methodshow.jl

```diff
@@ -119,8 +119,25 @@ end
 # In case the line numbers in the source code have changed since the code was compiled,
 # allow packages to set a callback function that corrects them.
 # (Used by Revise and perhaps other packages.)
-default_methodloc(method::Method) = method.file, method.line
-const methodloc_callback = Ref{Function}(default_methodloc)
+const methodloc_callback = Ref{Function}()
+
+# This function does the method location updating
+function updated_methodloc(m::Method)::Tuple{String, Int32}
+    file, line = string(m.file), m.line
+    if isassigned(methodloc_callback)
+        file, line = try
+            invokelatest(methodloc_callback[], m)
+        catch
+        end
```

Comment:
This will produce a misleading error if the call fails, won't it?
```julia
julia> a,b = try div(4,0) catch end
ERROR: MethodError: no method matching iterate(::Nothing)
Closest candidates are:
  iterate(::Core.SimpleVector) at essentials.jl:600
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:600
  iterate(::ExponentialBackOff) at error.jl:218
  ...
Stacktrace:
 [1] indexed_iterate(::Nothing, ::Int64) at .\tuple.jl:66
 [2] top-level scope at REPL[645]:1
```

Reply:
Yep, good point. I mixed two different things here (originally had something inside the `catch`).

Reply:
Idk how logging is handled for Base but I could imagine adding some `@debug e` (or even `@debug error(e)`) might help in some circumstances.